src/pages/dna/LoadedDna.tsx

```diff
@@ -65,13 +69,26 @@ const LoadedDna: FC<Props> = ({
     onSuccess: () => queryClient.invalidateQueries(getUserInfoBySurveyIdQueryKey(surveyId)),
   });
 
-  const onDownloadClick = () => {
+  const [isImageModalShowing, setIsImageModalShowing] = useState(false);
+
+  const onDownloadClick = async () => {
     const imageObj = JSON.parse(downloadableImageBase64);
     const imageBase64 = 'data:image/png;base64,' + imageObj.base64 ?? '';
-    // if (detectMobileDevice(window.navigator.userAgent)) {
-    //   return;
-    // }
-    imageDownloadPC(imageBase64, 'dna');
+    const browser = getBrowser();
+
+    // 1. 공유 모드가 가능하다면 공유 모드로 공유
+    if (typeof navigator.share !== 'undefined') {
+      if (await imageShare(imageBase64)) return;
+    }
```

Comment:
```suggestion
    if (typeof navigator.share !== 'undefined') {
      await imageShare(imageBase64)
      
      return;
    }
```

이렇게 해도 되는 거 같은데, 표현식을 if 문으로 감싸주신 이유가 있으신가요??

Reply:
imageShare 함수를 확인해보면 리턴값이 boolean인데, 혹시 실패하는 상황에는 return 하지 않음으로써 
후순위에 있는 이미지 다운 기능을 실행하기 위해 이렇게 작성하였습니다!

더 좋은 방향을 아신다면 알려주세요!

Reply:
오호 이미지 쉐어가 실패할 수도 있군뇨 몰랐어용
```tsx
const isImageShared = await imageShare(imageBase64);

if (isImageShare) return;
```

이렇게 풀어 쓸 수도 있을 거 같은데, 지금도 좋아 보입니다 ~~

Reply:
좋습니당~